android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java

```diff
@@ -94,41 +100,55 @@ public void onCreate() {
             .getSystemService(Context.BLUETOOTH_SERVICE);
         mBluetoothLeAdvertiser = bluetoothManager.getAdapter().getBluetoothLeAdvertiser();
         mNotificationManager = NotificationManagerCompat.from(this);
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+        mSharedPreferencesEditor = mSharedPreferences.edit();
```

Comment:
I'm not sure if starting edit mode begins a transaction, or only on your first edit.  Either way, don't create a member, just use this when you need it.